pennylane/debugging.py

```diff
@@ -53,18 +62,21 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         self.device._debugger = None
 
 
-def snapshots(qnode):
-    r"""Create a function that retrieves snapshot results from a QNode.
+@transform
+def snapshots(tape: QuantumTape) -> Tuple[Sequence[QuantumTape], Callable]:
+    r"""Transforms a QNode into several tapes by aggregating all operations up to a `qml.Snapshot`
+    operation into their own execution tape.
 
+    The output is a dictionary where each key is either the tag supplied to the snapshot or its
+    index in order of appearance, in additition to an "execution_results" that returns the final output
+    of the quantum circuit.
     Args:
-        qnode (.QNode): the input QNode to be simulated
+        tape (QNode or QuantumTape or Callable): a quantum circuit.
 
     Returns:
-        A function that has the same argument signature as ``qnode`` and returns a dictionary.
-        When called, the function will execute the QNode on the registered device and retrieve
-        the saved snapshots obtained via the :class:`~.pennylane.Snapshot` operation. Additionally,
-        the snapshot dictionary always contains the execution results of the QNode, so the use of
-        the tag "execution_results" should be avoided to prevent conflicting key names.
+        qnode (QNode) or quantum function (Callable) or tuple[List[QuantumTape], function]: The
+        transformed circuit as described in :func:`qml.transform <pennylane.transform>`.
+
```

Comment:
We should consider adding a `.. warning::` box cautioning the user about the fact that if the device doesn't natively support snapshots, this will perform one execution per snapshot.